client/js/lounge.js

```diff
@@ -675,50 +675,29 @@ $(function() {
 	});
 
 	chat.on("msg", ".messages", function(e, target, msg) {
-		var button = sidebar.find(".chan[data-target='" + target + "']");
+		if (msg.self) {
+			return;
+		}
+
+		var button = sidebar.find(".chan[data-target=" + target + "]");
+		var isActive = button.hasClass("active");
+		var type = msg.type;
+		var highlight = type.contains("highlight");
```

Comment:
Looks like you brought in older change which was removed, might be easier to just rewrite it on top of master manually.